examples/drop_sub_components.rs

```diff
@@ -0,0 +1,280 @@
+use std::fmt::Debug;
+
+use relm4::gtk::prelude::*;
+use relm4::prelude::*;
+use relm4::RelmRemoveAllExt;
+
+mod css {
+    pub const COMPONENT_SPACING: i32 = 5;
+}
+
+fn log_sender_output_error(sender: &str, message: impl Debug) {
+    eprintln!(
+        "Component {} failed to send output {:?} message",
+        sender, message
+    )
+}
+
+enum AppMode {
+    Initial(#[allow(dead_code)] Controller<InitialScreen>),
+    SubScreen1(#[allow(dead_code)] Controller<SubScreen1>),
+    SubScreen2(#[allow(dead_code)] Controller<SubScreen2>),
+}
+
+struct App {
+    mode: Option<AppMode>,
+}
+
+#[derive(Debug)]
+enum Msg {
+    ShowInitialScreen,
+    ShowSubScreen1,
+    ShowSubScreen2,
+}
+
+#[relm4::component]
+impl Component for App {
+    type Init = ();
+    type Input = Msg;
+    type Output = ();
+    type CommandOutput = ();
+
+    view! {
+        gtk::Window {
+            set_title: Some("Drop sub components"),
+            set_decorated: false,
+
+            gtk::Box {
+                set_orientation: gtk::Orientation::Vertical,
+                gtk::HeaderBar {},
+                #[name="container"]
+                gtk::Box {
+                    set_orientation: gtk::Orientation::Vertical,
+                }
+            }
+        }
+    }
+
+    fn init(
+        _init: Self::Init,
+        root: Self::Root,
+        sender: ComponentSender<Self>,
+    ) -> ComponentParts<Self> {
+        let mut model = Self { mode: None };
+        let mut widgets = view_output!();
+        model.update_with_view(&mut widgets, Msg::ShowInitialScreen, sender, &root);
+        ComponentParts { model, widgets }
+    }
+
+    fn update_with_view(
+        &mut self,
+        widgets: &mut Self::Widgets,
+        message: Self::Input,
+        sender: ComponentSender<Self>,
+        root: &Self::Root,
+    ) {
+        widgets.container.remove_all();
+        match message {
+            Msg::ShowInitialScreen => {
+                let controller =
+                    InitialScreen::builder()
+                        .launch(())
+                        .forward(sender.input_sender(), |msg| match msg {
+                            InitialScreenOutput::ShowSubScreen1 => Msg::ShowSubScreen1,
+                            InitialScreenOutput::ShowSubScreen2 => Msg::ShowSubScreen2,
+                        });
+                widgets.container.append(controller.widget());
```

Comment:
Does that work as expected? I think the component runtime will be dropped as expected, but the widgets will remain in the `Box`. Using [`CenterBox::set_center_widget`](https://gtk-rs.org/gtk4-rs/git/docs/gtk4/struct.CenterBox.html#method.set_center_widget) would automatically remove the widgets as well.